Content.Client/CrewManifest/UI/CrewManifestListing.cs

```diff
@@ -21,32 +22,66 @@ public CrewManifestListing()
+        // create an inverse role -> department lookup
+        var jobDepartments = new Dictionary<ProtoId<JobPrototype>, List<DepartmentPrototype>>();
+        foreach (var department in _prototypeManager.EnumeratePrototypes<DepartmentPrototype>())
         {
-            foreach (var department in _prototypeManager.EnumeratePrototypes<DepartmentPrototype>())
+            foreach (var roleId in department.Roles.Where(roleId => _prototypeManager.HasIndex(roleId)))
             {
-                // this is a little expensive, and could be better
-                if (department.Roles.Contains(entry.JobPrototype))
-                {
-                    entryDict.GetOrNew(department).Add(entry);
-                }
+                jobDepartments.GetOrNew(roleId).Add(department);
             }
         }
 
-        var entryList = new List<(DepartmentPrototype section, List<CrewManifestEntry> entries)>();
-
-        foreach (var (section, listing) in entryDict)
+        // still sorted non-marine entries grouped by department
+        var departmentDict = new Dictionary<DepartmentPrototype, List<CrewManifestEntry>>();
+        // still sorted marine entries grouped by squad
+        var squadDict = new Dictionary<string, List<CrewManifestEntry>>();
+        foreach (var entry in entries.Entries)
         {
-            entryList.Add((section, listing));
+            if (!jobDepartments.TryGetValue(entry.JobPrototype, out var cachedDeps))
+            {
+                // this shouldn't happen. maybe throw instead?
+                continue;
+            }
+
+            // squad role?
+            if (!string.IsNullOrEmpty(entry.Squad))
+            {
+                squadDict.GetOrNew(entry.Squad).Add(entry);
+                continue;
+            }
+
+            foreach (var department in cachedDeps)
+            {
+                departmentDict.GetOrNew(department).Add(entry);
+            }
         }
 
-        entryList.Sort((a, b) => DepartmentUIComparer.Instance.Compare(a.section, b.section));
+        // sort by department weight
+        foreach (var (department, depEntries) in departmentDict
+                     .OrderBy(kvp => kvp.Key, DepartmentUIComparer.Instance))
+        {
+            AddChild(new CrewManifestSection(
+                _prototypeManager,
+                _spriteSystem,
+                Loc.GetString(department.Name),
+                depEntries));
+        }
 
-        foreach (var item in entryList)
+        // sort by squad name
+        foreach (var (squad, squadEntries) in squadDict
+                     .OrderBy(kvp => kvp.Key))
         {
-            AddChild(new CrewManifestSection(_prototypeManager, _spriteSystem, item.section, item.entries));
+            // server already sorts by job weight > name
+            AddChild(new CrewManifestSection(
+                _prototypeManager,
+                _spriteSystem,
+                Loc.GetString(squad),
+                squadEntries));
```

Comment:
This code should be put at the top of the file and then separated from the original with a return; for merge conflicts

Reply:
i separated the changes as best as i understood your requirements. hope this is enough

p.s. funny remark: for 3 months of being inactive this pr had no merge conflicts